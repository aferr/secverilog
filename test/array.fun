(declare-fun Par (Int) Label)
(assert (= (Par 0) LOW))
(assert (= (Par 1) HIGH))
(assert (= (Par 2) HIGH))
