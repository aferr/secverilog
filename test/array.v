
module array(clk, reset, sel);

   input {L} clk;
   input {L} reset;
   input {L} sel;
   
   wire [15:0] {H} sec_data_val;
   wire [3:0]  {H} sec_index_val;

   wire [15:0] {L} pub_data_val;
   wire [3:0]  {L} pub_index_val;
   
   reg [15:0] seq {L} data_array[3:0];
   reg [15:0] {L} data_array_com[3:0];


   always@(posedge clk) begin
      if (sel) begin
	 data_array[pub_index_val] <= pub_data_val; //success
         data_array[sec_index_val] <= pub_data_val; //fail
	 data_array[0][3:0] <= pub_index_val; //success
	 data_array[0][sec_index_val] <= pub_index_val; //fail	 
      end else begin
	 data_array[pub_index_val] <= sec_data_val; //fail
	 data_array[sec_index_val] <= sec_data_val; //fail
	 data_array[0][3:0] <= sec_index_val; //fail
	 data_array[0][sec_index_val] <= sec_index_val; //fail	 	 
      end
   end


   always@(*) begin
      if (sel) begin
	 data_array_com[pub_index_val] = pub_data_val; //success
         data_array_com[sec_index_val] = pub_data_val; //fail
	 data_array_com[0][3:0] = pub_index_val; //success
	 data_array_com[0][sec_index_val] = pub_index_val; //fail	 
      end else begin
	 data_array_com[pub_index_val] = sec_data_val; //fail
	 data_array_com[sec_index_val] = sec_data_val; //fail
	 data_array_com[0][3:0] = sec_index_val; //fail
	 data_array_com[0][sec_index_val] = sec_index_val; //fail	 	 
      end
   end
   
  

   
endmodule // array
