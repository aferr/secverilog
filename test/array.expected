base conditions are satisfiable? (should be sat)
sat
unsat
sat
unsat
sat
sat
sat
sat
sat
unsat
sat
unsat
sat
sat
sat
sat
sat
unsat
